submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,239 @@
+ovn-k8s-mp0 interface to reach the host networking stack and then the Submariner tunnel. This interface is used by OVN for host-networking
+traffic with in a cluster and will be present in every node.
+
+![UpdatedTopology](./images/submariner-ovn-ic.png)
+
+### SubmarinerRouteAgentPod
+
+The Submariner Route-agent pod will be responsible for creating the SubmarinerGWRoute CR. It will be used only for OVN CNI right now. For every
+RemoteEndpointCreated event a SubmarinerGWRouteCR will be created. The nextHop will be the interface IP through which we can reach the cable
+driver. In the case of OVN it will be the IP of ovn-k8s-mp0 interface.
+
+The SubmarinerDefaultRoute CRD will also be created by Submariner Route-agent. It will have the list of remote endpoints connected to the gateway.
+The nexthop will be the transit switch IP of the G/W node. If the transit switch IP is missing this CRD will not be created, which means it is
+a non-IC setup.
+
+The Routeagent will have these controllers added to it
+
+#### SubmarinerGWRouteCR Controller
+
+This controller will be responsible for programming the OVN cluster router, it will react only in a g/w node. When a SubmarinerGWRoute
+CR is created or modified the controller shall create or update a routing policy in OVN cluster router with a priority of 20000, and it should
+redirect any traffic destined to remote CIDR to the ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : 0459f009-3603-47ac-8ee7-9d958540ed31
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.244.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+##### SubmarinerNonGWRouteController
+
+This controller will run in every route agent pod. This controller connects to the OVN DB. When a SubmarinerDefaultRoute CR is created
+
+* non-gateway node : it updates the DB with a router policy with a priority 20000 to send the traffic to the remote cluster via next hop
+  mentioned, which is the transit switch IP to the g/w node. Before adding the route it checks if a route exists, if so it skips adding the
+  route again. This is required to prevent duplicate update since there can be more than one node in each zone and hence more than one
+  Routeagent.
+
+```bash
+_uuid               : 22db3005-64c5-4e32-aeb0-642423c30742
+action              : reroute
+external_ids        : {}
+match               : "ip4.dst==10.132.0.0/16"
+nexthop             : []
+nexthops            : ["169.254.0.1"]
+options             : {"external_ids:{submariner"="true}"}
+priority            : 20000
+```
+
+* gateway node/same zone node : In the gateway node route agent adds a route, if it does not exist, to send the traffic coming from other
+  zones which is destined to remote cluster IP range to the ovn-k8s-mp0 interface IP. It checks if the rule already exists to prevent duplicate
+  update.
```

Comment:
```suggestion
* gateway node/same zone node : Route agent running on the Gateway node, programs a route in the ovn-cluster-router, to route the traffic coming from other zones destined to remote cluster IP range via the ovn-k8s-mp0 interface IP. It checks if the rule already exists to prevent duplicate update.
```